src/storage/hummock_sdk/src/compaction_group/mod.rs

```diff
@@ -46,11 +46,217 @@ impl From<StaticCompactionGroupId> for CompactionGroupId {
+    }
+
+    /// generate a full key for convenience to get the `table_id` and `vnode`
+    pub fn build_split_full_key(
+        mut table_id: StateTableId,
+        mut vnode: VirtualNode,
+    ) -> FullKey<Vec<u8>> {
+        if VirtualNode::MAX == vnode {
+            // Modify `table_id` to `next_table_id` to satisfy the `split_to_right`` rule, so that the `table_id`` originally passed in will be split to left.
+            table_id = table_id.strict_add(1);
+            vnode = VirtualNode::ZERO;
+        }
+
+        FullKey::new(
+            TableId::from(table_id),
+            TableKey(vnode.to_be_bytes().to_vec()),
+            HummockEpoch::MAX,
+        )
+    }
+
+    #[derive(Debug, PartialEq, Clone)]
+    pub enum SstSplitType {
+        Left,
+        Right,
+        Both,
+    }
+
+    /// Determine whether the SST needs to be split, and if so, which side to split.
+    pub fn need_to_split(sst: &SstableInfo, split_key: Bytes) -> SstSplitType {
+        let key_range = &sst.key_range;
+        // 1. compare left
+        if KeyComparator::compare_encoded_full_key(&split_key, &key_range.left).is_le() {
+            return SstSplitType::Right;
+        }
+
+        // 2. compare right
+        if key_range.right_exclusive {
+            if KeyComparator::compare_encoded_full_key(&split_key, &key_range.right).is_ge() {
+                return SstSplitType::Left;
+            }
+        } else if KeyComparator::compare_encoded_full_key(&split_key, &key_range.right).is_gt() {
+            return SstSplitType::Left;
+        }
+
+        SstSplitType::Both
+    }
 
+    /// Split the SST into two parts based on the split key.
+    /// The left part is the original SST, and the right part is the new SST.
+    /// The split key is exclusive for the left part and inclusive for the right part.
+    /// The `table_ids` of the new SST are calculated based on the split key.
+    /// e.g.
+    ///  `sst.table_ids` = [1, 2, 3, 4, 5, 6, 7, 8, 9], `split_key` = (`table_id` = 5, vnode = 0)
+    ///  then the result:
+    /// sst1 {
+    ///     `sst_id`: `new_sst_id + 1`,
+    ///     `table_ids`: [1, 2, 3, 4],
+    ///     `key_range`: [left, `split_key`),
+    ///     `sst_size`: `left_size`,
+    /// }
+    /// sst2 {
+    ///    `sst_id`: `new_sst_id`,
+    ///    `table_ids`: [5, 6, 7, 8, 9],
+    ///    `key_range`: [`split_key`, right],
+    ///    `sst_size`: `right_size`,
+    /// }
+    pub fn split_sst(
+        origin_sst_info: &mut SstableInfo,
+        new_sst_id: &mut u64,
+        split_key: Bytes,
+        left_size: u64,
+        right_size: u64,
+    ) -> SstableInfo {
+        let mut branch_table_info = origin_sst_info.clone();
+        branch_table_info.sst_id = *new_sst_id;
+        *new_sst_id += 1;
+        origin_sst_info.sst_id = *new_sst_id;
+        *new_sst_id += 1;
+
+        let (key_range_l, key_range_r) = {
+            let key_range = &origin_sst_info.key_range;
+            let l = KeyRange {
```

Comment:
In the context of split_sst, the relative ordering of sst.left,sst.right and split_key cannot be assumed.
Maybe assert l.left < l.right and r.left<=r.right, or handle it properly?

Reply:
e.g. split_key = sst.left = sst.right (inclusive).
We don't want any resulted sst.left > sst.right.

Reply:
Good catch, refactor `split_sst` and add related UT